source/type/vehicle_journey.cpp

```diff
@@ -324,7 +332,7 @@ std::set<uint16_t> VehicleJourney::get_sections_ranks(const StopArea* start_sa,
 
     // Must close a potential section
     if (section_ending) {
-        for (uint16_t i = *section_start_rank; i <= vj->stop_time_list.size() - 1; ++i) {
+        for (auto i = *section_start_rank; i <= RankStopTime(vj->stop_time_list.size() - 1); ++i) {
```

Comment:
Don't know if the readability is better now, if you have no knowledge in this part of code. You see what I mean ?
`vj->stop_time_list.size() - 1` is already a size, so why it is better to have `RankStopTime(vj->stop_time_list.size() - 1)`

Reply:
I change it to avoid `- 1` for readability, cleanness and signed/unsigned issue (FYI no clang warning were raised).
But it's nice to keep the `Rank` type as it's inserted below, so I'll keep it for now.
Let me know if you think of a better alternative.

Reply:
Okay, weirdo for the `auto`. Modern compiler must manage `auto` easyly with deduction. 
Good to know